crates/nu-plugin-engine/src/persistent.rs

```diff
@@ -75,6 +77,7 @@ impl PersistentPlugin {
     pub fn get(
         self: Arc<Self>,
         envs: impl FnOnce() -> Result<HashMap<String, String>, ShellError>,
+        ctrlc_handlers: Option<ctrlc::Handlers>,
```

Comment:
Since this is always the same, I think it can be added when the `PersistentPlugin` is constructed instead of at `get()` time. It should never change - using a different value here would be unpredictable because it would depend on whether the plugin has already been spawned or not, so if we want to fork that handling in some way we'll have to do it differently anyway.

Reply:
An EngineState isn't currently needed to construct a PersistentPlugin, so there's nothing that links a plugin and an engine_state until PersistentPlugin.get_plugin is called. I wonder if we should separate getting and spawning a plugin? `envs()` is similar in that it's only past to `get` in case `get` needs to spawn, but it's otherwise ignored.

Reply:
It's a part of `EngineState`, but it doesn't require a whole `EngineState`. I don't see a problem with it. You can always pass `None`.

Reply:
I think I'm not understanding the need for the `GetPlugin` trait? could we put the envs closure on PersistentPlugin as well?